utils/file.go

```diff
@@ -29,3 +32,39 @@ func FileExists(filename string) bool {
 	_, err := os.Stat(filename)
 	return err == nil
 }
+
+// ShebangLine extracts the shebang line from the file.
+func ShebangLine(path string) (string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer f.Close()
+	sc := bufio.NewScanner(f)
+	if !sc.Scan() {
+		// If the scan ended because of EOF, the file is empty, so sc.Err = nil.
+		// Otherwise sc.Err reflects the error reading the file.
+		return "", sc.Err()
+	}
+	line := sc.Text()
+	if !strings.HasPrefix(line, "#!") {
+		// Not a shebang line
+		return "", nil
+	}
+	return line, nil
+}
+
+// IsPOSIXShell attempts to detect POSIX-compliant shells (e.g bash, sh, zsh)
+func IsPOSIXShell(shell []string) bool {
+	bin := filepath.Base(shell[0])
+	if bin == "env" {
+		bin = filepath.Base(shell[1])
```

Comment:
If the shebang were just `#!/usr/bin/env` (which is an error that would have been silently ignored before), I think we will panic here. So explicit bounds checking might be necessary.

Reply:
Done.